src/Core/System/SystemConfig/Subscriber/MemoizedSystemConfigLoaderResetSubscriber.php

```diff
@@ -0,0 +1,69 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\SystemConfig\Subscriber;
+
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+use Shopware\Core\System\SystemConfig\AbstractSystemConfigLoader;
+use Shopware\Core\System\SystemConfig\Event\SystemConfigChangedEvent;
+use Shopware\Core\System\SystemConfig\MemoizedSystemConfigLoader;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class MemoizedSystemConfigLoaderResetSubscriber implements EventSubscriberInterface, ResetInterface
+{
+    private AbstractSystemConfigLoader $systemConfigLoader;
+
+    public function __construct(AbstractSystemConfigLoader $systemConfigLoader)
+    {
+        $this->systemConfigLoader = $systemConfigLoader;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SystemConfigChangedEvent::class => [
+                ['onValueChanged', 1500],
+            ],
+        ];
+    }
+
+    public function reset(): void
+    {
+        $systemConfigLoader = $this->getMemoizedSystemConfigLoader();
+
+        if ($systemConfigLoader === null) {
+            return;
+        }
+
+        $systemConfigLoader->reset();
+    }
+
+    public function onValueChanged(SystemConfigChangedEvent $event): void
+    {
+        $systemConfigLoader = $this->getMemoizedSystemConfigLoader();
+
+        if ($systemConfigLoader === null) {
+            return;
+        }
+
+        $salesChannelId = $event->getSalesChannelId();
+
+        $systemConfigLoader->resetForSalesChannelId($salesChannelId);
+    }
+
+    private function getMemoizedSystemConfigLoader(): ?MemoizedSystemConfigLoader
+    {
+        $service = $this->systemConfigLoader;
+
+        try {
+            do {
+                if ($service instanceof MemoizedSystemConfigLoader) {
+                    return $service;
+                }
+            } while ($service = $service->getDecorated());
```

Comment:
How would you do that with only an interface :trollface: 

refering to https://github.com/shopware/platform/issues/2347 :wink: